src/khoj/database/migrations/0035_convert_png_to_webp.py

```diff
@@ -0,0 +1,77 @@
+# Generated by Django 4.2.10 on 2024-04-13 17:54
+
+import base64
+import io
+
+from django.db import migrations
+from PIL import Image
+
+
+def convert_png_images_to_webp(apps, schema_editor):
+    # Get the model from the versioned app registry to ensure the correct version is used
+    Conversations = apps.get_model("database", "Conversation")
+    for conversation in Conversations.objects.all():
+        for chat in conversation.conversation_log["chat"]:
+            if chat["by"] == "khoj" and chat["intent"]["type"] == "text-to-image":
+                # Decode the base64 encoded PNG image
+                decoded_image = base64.b64decode(chat["message"])
+
+                # Convert images from PNG to WebP format
+                image_io = io.BytesIO(decoded_image)
+                with Image.open(image_io) as png_image:
+                    webp_image_io = io.BytesIO()
+                    png_image.save(webp_image_io, "WEBP")
+
+                    # Encode the WebP image back to base64
+                    webp_image_bytes = webp_image_io.getvalue()
+                    chat["message"] = base64.b64encode(webp_image_bytes).decode()
+                    chat["intent"]["type"] = "text-to-image-v3"
+                    webp_image_io.close()
+
+            if chat["by"] == "khoj" and chat["intent"]["type"] == "text-to-image2":
+                print("❗️ Please MANUALLY update PNG images created by Khoj in your AWS S3 bucket to WebP format.")
+                # Convert PNG url to WebP url
+                chat["message"] = chat["message"].replace(".png", ".webp")
```

Comment:
I think we can just leave these as pngs, as self-hosted folks probably won't go through the effort of migrating & would remain on the `text-to-image2` type. We don't even really need to output the warning, the end user shouldn't have to worry about the underlying image type. 

You can create a custom management command for anyone self-hosting who might want to migrate to run this operation: https://docs.djangoproject.com/en/5.0/howto/custom-management-commands/.

Reply:
Good point. Created, tested the management command to ensure it works